projects/step-core/src/lib/directives/test-dataid.directive.ts

```diff
@@ -0,0 +1,14 @@
+import { Directive, ElementRef, inject, Input } from '@angular/core';
+
+@Directive({
+  selector: '[testDataId]',
+})
+export class TestDataIdDirective {
+  @Input('testDataId') testDataId: string | undefined;
```

Comment:
I would add the behavior, that this identifier might change on the fly.
Even in most cases it will be a constant string, ability to react on this input change might prevent some future errors.
input signal + effect, instead of ngOnInit will look nice in this case